src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -659,6 +660,26 @@ internal static bool WriteConsideredUse(TypeSymbol type, BoundExpression value)
             }
         }
 
+        private static bool IsConstant(BoundExpression value)
```

Comment:
It feels like this should be a local function. In particular I wouldn't want someone to discover this and think that returning `true` means the BoundExpression is a constant in the [language sense](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#constant-expressions).

Also, it feels like we could consider making a spec change so that interpolated strings whose inserts are all constant values can be considered themselves to be constant values.

Reply:
Yes, we should consider allowing interpolated strings with no substitutions as constants. Let's follow up with language design.

---
In reply to: [344319089](https://github.com/dotnet/roslyn/pull/39734#discussion_r344319089) [](ancestors = 344319089)